src/lib/eo/eo_ptr_indirection.c

```diff
@@ -39,9 +39,9 @@ _eo_obj_pointer_invalid(const Eo_Id obj_id,
    const char *reason = "This ID has probably been deleted";
    char tbuf[128];
    if (obj_id & MASK_CLASS_TAG) type = "class";
-   if (thread != _efl_object_main_thread)
+   if (!eina_thread_equal(thread, _efl_object_main_thread))
      {
-        snprintf(tbuf, sizeof(tbuf), "%p", (void *)thread);
```

Comment:
Maybe we could do something like:
```C
   Eina_Thread thread = eina_thread_self();
   Eina_ThreadId thread_id = eina_thread_self_id();
   ...
   if (!eina_thread_equal(thread, _efl_object_main_thread))
     {
        snprintf(tbuf, sizeof(tbuf), "%ld", (long) thread_id);
...
```
I'm just not sure if that cast will work fine with a `DWORD`, `int`, `pid_t` and `size_t`

Reply:
Is this line even a problem? I mean, it only prints the pointer right? Does it matter how the data is defined?

Reply:
It's used for logging. I think it's important to print out a value that helps developers with debugging.

Reply:
I aggree with getting the thread_id explicitly. It's more righteous with expecting the ID to be shown, and not _insert abstract data-type here_.

Reply:
Done: 06208c2